rules-java-diva/tests/src/test/java/org/jboss/windup/rules/apps/diva/DivaTest.java

```diff
@@ -0,0 +1,116 @@
+            @AddonDependency(name = "org.jboss.windup.rules.apps:windup-rules-java"),
+            @AddonDependency(name = "org.jboss.forge.furnace.container:cdi")
+    })
+    public static AddonArchive getDeployment() {
+        return ShrinkWrap.create(AddonArchive.class).addBeansXML();
+    }
+
+    @Inject
+    private WindupProcessor processor;
+
+    @Inject
+    private GraphContextFactory factory;
+
+    @Test
+    public void testSpring() throws IOException {
+        try (GraphContext context = factory.create(true)) {
+            Path inputPath = Paths.get("src/test/resources/spring/");
+            Path outputPath = Paths.get(FileUtils.getTempDirectory().toString(), "Windup")
+                    .resolve(UUID.randomUUID().toString());
+
+            WindupConfiguration windupConfiguration = new WindupConfiguration().setGraphContext(context)
+                    .setOptionValue(SourceModeOption.NAME, true).addInputPath(inputPath).setOutputDirectory(outputPath);
+
+            processor.execute(windupConfiguration);
+
+            List<? extends ProjectModel> projects = IterableUtils.toList(context.findAll(DivaAppModel.class));
+            Assert.equals(projects.size(), 1);
+            List<? extends DivaTxModel> txs = IterableUtils.toList(context.findAll(DivaTxModel.class));
+            Assert.equals(txs.size(), 1);
+            List<?> ops = context.getQuery(DivaOpModel.class).getRawTraversal()
+                    .has("sql", P.without("BEGIN", "COMMIT", "ROLLBACK")).toList();
+            Assert.equals(ops.size(), 2);
+
+        }
+    }
+
+    @Test
+    public void testServlet() throws IOException {
+        try (GraphContext context = factory.create(true)) {
+            Path inputPath = Paths.get("src/test/resources/servlet/");
+            Path outputPath = Paths.get(FileUtils.getTempDirectory().toString(), "Windup")
+                    .resolve(UUID.randomUUID().toString());
+
+            WindupConfiguration windupConfiguration = new WindupConfiguration().setGraphContext(context)
+                    .setOptionValue(SourceModeOption.NAME, true).addInputPath(inputPath).setOutputDirectory(outputPath);
+
+            processor.execute(windupConfiguration);
+
+            List<? extends ProjectModel> projects = IterableUtils.toList(context.findAll(DivaAppModel.class));
+            Assert.equals(projects.size(), 1);
+            List<? extends DivaTxModel> txs = IterableUtils.toList(context.findAll(DivaTxModel.class));
+            Assert.equals(txs.size(), 1);
+            List<?> ops = context.getQuery(DivaOpModel.class).getRawTraversal()
+                    .has("sql", P.without("BEGIN", "COMMIT", "ROLLBACK")).toList();
+            Assert.equals(ops.size(), 1);
+        }
+    }
+
+    @Ignore
+    @Test
+    public void testDayTrader() throws IOException {
+        try (GraphContext context = factory.create(true)) {
+            Path inputPath = Paths.get("../../../sample.daytrader7");
```

Comment:
This path must be changed sooner or later to have the test to be self-container without relying on external resources.